Wrappers/Python/ccpi/optimisation/operators/BlockOperator.py

```diff
@@ -0,0 +1,744 @@
+    # __radd__
+    
+    def __rsub__(self, other):
+        return (-1 * self) + other
+    # __rsub__
+    
+    def __rmul__(self, other):
+        '''Reverse multiplication
+        
+        to make sure that this method is called rather than the __mul__ of a numpy array
+        the class constant __array_priority__ must be set > 0
+        https://docs.scipy.org/doc/numpy-1.15.1/reference/arrays.classes.html#numpy.class.__array_priority__
+        '''
+        return self * other
+    # __rmul__
+    
+    def __rdiv__(self, other):
+        return pow(self / other, -1)
+    # __rdiv__
+    def __rtruediv__(self, other):
+        return self.__rdiv__(other)
+    
+    def __rpow__(self, other):
+        return other.power(self)
+    
+    def __iadd__(self, other):
+        if isinstance (other, BlockDataContainer):
+            for el,ot in zip(self.containers, other.containers):
+                el += ot
+        elif isinstance(other, Number):
+            for el in self.containers:
+                el += other
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            self.is_compatible(other)
+            for el,ot in zip(self.containers, other):
+                el += ot
+        return self
+    # __radd__
+    
+    def __isub__(self, other):
+        if isinstance (other, BlockDataContainer):
+            for el,ot in zip(self.containers, other.containers):
+                el -= ot
+        elif isinstance(other, Number):
+            for el in self.containers:
+                el -= other
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            assert self.is_compatible(other)
+            for el,ot in zip(self.containers, other):
+                el -= ot
+        return self
+    # __rsub__
+    
+    def __imul__(self, other):
+        if isinstance (other, BlockDataContainer):
+            for el,ot in zip(self.containers, other.containers):
+                el *= ot
+        elif isinstance(other, Number):
+            for el in self.containers:
+                el *= other
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            assert self.is_compatible(other)
+            for el,ot in zip(self.containers, other):
+                el *= ot
+        return self
+    # __imul__
+    
+    def __idiv__(self, other):
+        if isinstance (other, BlockDataContainer):
+            for el,ot in zip(self.containers, other.containers):
+                el /= ot
+        elif isinstance(other, Number):
+            for el in self.containers:
+                el /= other
+        elif isinstance(other, list) or isinstance(other, numpy.ndarray):
+            assert self.is_compatible(other)
+            for el,ot in zip(self.containers, other):
+                el /= ot
+        return self
+    # __rdiv__
+    def __itruediv__(self, other):
+        return self.__idiv__(other)
+    
+
+       
+class BlockOperator(Operator):
```

Comment:
I think implicitly we have considered this is a linear operator, since we have `adjoint` method present and also due to the linearity being used in the `direct` and `adjoint` methods. Maybe name should better be `BlockLinearOperator`?